openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIHeaderMapper.java

```diff
@@ -67,9 +76,21 @@ public List<Parameter> setHeaderParameter(RequiredParameterNode headerParam) {
         String headerName = unescapeIdentifier(extractHeaderName(headerParam));
         HeaderParameter headerParameter = new HeaderParameter();
         Node node = headerParam.typeName();
-        Schema<?> headerTypeSchema = ConverterCommonUtils.getOpenApiSchema(getHeaderType(headerParam));
+        Schema<?> headerTypeSchema;
+        String isOptional = Constants.FALSE;
```

Comment:
is there any reason why we cant make `isOptional` a boolean flag?

Reply:
Mainly there is a scenario where we have to get optional details by extracting the `@http:ServiceConfig` annotation and it gives the String type details. That is the reason we make it a string variable.